R/add_lines.R

```diff
@@ -0,0 +1,25 @@
+#' Add lines
+#'
+#' @param data dataset to divide into lines
+#' @param extract_length total length of extract in ms
+#' @param window_size length of line in ms
+#'
+#' @return data set with 'line' column added
+#' @export
+#'
+add_lines <- function(data,extract_length=600000,window_size=60000) {
+
+  window_breaks <- as.integer(c(0:round(extract_length/window_size)) * window_size)
+
+  data <- data |>
+    dplyr::mutate(end = end - min(begin), # reset timestamps to start from 0
+           begin = begin - min(begin),
+           line = cut(begin,window_breaks,right=F,labels=F)) |>
+    tidyr::drop_na(line) |>
+    dplyr::group_by(line) |>
+    dplyr::mutate(begin0 = begin - min(begin), # reset timestamps to 0 for each new line
+           end0 = end - min(begin)) |>
```

Comment:
The new columns are named `begin0` and `end0`. Maybe it is more clearly associated to the line when they are labeled differently; eg `begin_line`, or `line_relative_begin`...?

Reply:
I like it short, so `begin_line`, `end_line`